File: android/app/src/main/java/dr/achim/sleep_timer/CallbackHelper.java
package dr.achim.sleep_timer;

import android.content.Context;
import android.content.SharedPreferences;

public class CallbackHelper {
    static String ENTRY_POINT_HELPER_PREF = "SLEEP_TIMER_CALLBACK_PREF";
    static String CALLBACK_HANDLE_KEY = "SLEEP_TIMER_CALLBACK_KEY";

    static Long NO_HANDLE = -1L;

    static void setHandle(Context context, Long handle) {
        final SharedPreferences.Editor editor= context.getSharedPreferences(
                ENTRY_POINT_HELPER_PREF,
                Context.MODE_PRIVATE
        ).edit();

        editor.putLong(CALLBACK_HANDLE_KEY, handle);
        editor.apply();
    }

    static Long getRawHandle(Context context) {
        return context.getSharedPreferences(
                ENTRY_POINT_HELPER_PREF,
                Context.MODE_PRIVATE
        ).getLong(CALLBACK_HANDLE_KEY, NO_HANDLE);
    }
}

File: android/app/src/main/kotlin/dr/achim/sleep_timer/CallbackHelper.kt
package dr.achim.sleep_timer

import android.content.Context

object CallbackHelper {
    const val ENTRY_POINT_HELPER_PREF = "SLEEP_TIMER_CALLBACK_PREF"
    const val CALLBACK_HANDLE_KEY = "SLEEP_TIMER_CALLBACK_KEY"
    const val NO_HANDLE = -1L
    fun setHandle(context: Context, handle: Long?) {
        val editor = context.getSharedPreferences(
                ENTRY_POINT_HELPER_PREF,
                Context.MODE_PRIVATE
        ).edit()
        editor.putLong(CALLBACK_HANDLE_KEY, handle!!)
        editor.apply()
    }

    fun getRawHandle(context: Context): Long {
        return context.getSharedPreferences(
                ENTRY_POINT_HELPER_PREF,
                Context.MODE_PRIVATE
        ).getLong(CALLBACK_HANDLE_KEY, NO_HANDLE)
    }
}